File: Ashutosh-Singh/SlidingWindow/Alternating-Groups-I.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int numberOfAlternatingGroups(vector<int>& colors) {
        int n = colors.size();
        int count = 0;

        for (int i = 0; i < n; ++i) {
            int prev = colors[(i - 1 + n) % n];
            int curr = colors[i];
            int next = colors[(i + 1) % n];

            if (prev != curr && curr != next && prev == next) {
                ++count;
            }
        }

        return count;
    }
};

int main() {
    int n;
    cin >> n;
    vector<int> colors(n);

    for (int i = 0; i < n; ++i) {
        cin >> colors[i];
    }

    Solution solution;
    cout << solution.numberOfAlternatingGroups(colors) << endl;

    return 0;
}

File: Shantanu_Mamgain/Sliding-Window-Leetcode/Alternating Groups I.java
class Solution {
    public int numberOfAlternatingGroups(int[] colors) {
        int start = 0;
        int end = 2;
        int count = 0;
        while(start < colors.length) {
            if(colors[start] == colors[end] && colors[(start+1)%colors.length] != colors[start]){
                count++;
            }
            start ++;
            end = (end+1)%colors.length;
        }
        return count;
    }

}
